mlx_lm/models/lfm2_moe.py

```diff
@@ -192,8 +191,8 @@ def __call__(
         self,
         x: mx.array,
     ):
-        gates = self.gate(x)
-        gates = mx.softmax(gates, axis=-1, precise=True)
+        gates = self.gate(x).astype(mx.float32)
```

Comment:
Why do we upcast it to float32 here?

Reply:
This score computation in MoEs is usually quite sensitive to precision, especially with a bias which is kept in fp32 (and it is also not that expensive).

Reply:
That makes sense.

Thanks for clarifying!